packages/medusa/src/api/routes/admin/products/update-product.ts

```diff
@@ -381,6 +396,14 @@ export class AdminPostProductsProductReq {
   @IsArray()
   tags?: ProductTagReq[]
 
+  @FeatureFlagDecorators(SalesChannelFeatureFlag.key, [
```

Comment:
should we also validate the negative, i.e if the featureflag is not enabled the `sales_channels` property is required to be undefined?